rocAL/rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -405,12 +403,16 @@ rocalSequenceReaderSingleShard(
 
         INFO("Internal buffer size width = "+ TOSTR(width)+ " height = "+ TOSTR(height) + " depth = "+ TOSTR(num_of_planes))
 
-        auto info = ImageInfo(width, height,
-                              context->internal_batch_size(),
-                              num_of_planes,
-                              context->master_graph->mem_type(),
-                              color_format );
-        output = context->master_graph->create_loader_output_image(info);
+        std::vector<size_t> dims = {context->user_batch_size(), sequence_length, height, 
+                                    width, static_cast<unsigned>(num_of_planes)};
+        auto info  = rocalTensorInfo(std::move(dims),
+                                     context->master_graph->mem_type(),
+                                     RocalTensorDataType::UINT8);
+        info.set_color_format(color_format);
+        info.set_tensor_layout(RocalTensorlayout::NFHWC);
+        info.set_sequence_batch_size(sequence_length);
+        info.set_max_shape();
```

Comment:
what does set_max_shape() api do here?

Reply:
Set max shape is to store the max width and height of the tensor wrt the layout passed.
It is set only only for the dataloaders i.e the first node.